Silicon/ApollolakePkg/Library/FirmwareUpdateLib/FirmwareUpdateLib.c

```diff
@@ -508,5 +511,8 @@ ClearFwUpdateTrigger (
   VOID
   )
 {
+  // Clear platform firmware update trigger.
+  MmioAnd32 (PMC_BASE_ADDRESS + R_PMC_BIOS_SCRATCHPAD, 0xFF00FFFF);

```

Comment:
Instead of PMC_BASE_ADDRESS,
PmcBar = MmioRead32 (MM_PCI_ADDRESS (0, PCI_DEVICE_NUMBER_PMC, PCI_FUNCTION_NUMBER_PMC, R_PMC_BASE)) & ~0x0F;

Reply:
PMC device might be hidden.   But the BAR is programmed inside FSP-T.   
Since FSP used fixed base for PMC base (not resource reallocation).  I guess it is fine to do this.

Reply:
Got it. I am okay with this. Thanks.

Reply:
thx!